File: monitoring-backend/src/main/java/com/performance/monitoring/models/Metric.java
package com.performance.monitoring.models;

import java.util.Date;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "Metrics")
public class Metric {
    @Id
    private String id;
    private String name;
    private int value;
    private String date;
    private int position;
    private String taskId;
    
    public Metric(String id, String name, int value, String date, int position, String taskId) {
        this.id = id;
        this.name = name;
        this.value = value;
        this.date = date;
        this.position = position;
        this.taskId = taskId;
    }

    public String getTaskId() {
        return taskId;
    }

    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public int getValue() {
        return value;
    }

    public void setValue(Integer value) {
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}

File: monitoring-frontend/src/models/metric.ts
export class Metric {
    id: string;
    name: string;
    value: number;
    date: string;
    position: number
    taskId: string;

    constructor(id: string, name: string, value: number, date: string, position: number, taskId: string) {
        this.id = id;
        this.name = name;
        this.value = value;
        this.position = position;
        this.date = date;
        this.taskId = taskId;
    }
}

export class MetricList {
    metrics: Metric[];
    constructor(metrics: Metric[]) {
        this.metrics = metrics;
    }
}
